### PR TITLE
hack/cherry_pick_pull.sh: Coach user on how to make a pull against non-master

### PR DESCRIPTION
**Pull request by zmerlynn**

### CODE REVIEW

**Comment by zmerlynn** [comment]
cc @thockin @davidopp

**Comment by thockin** [comment]
LGTM

@davidopp can mark it OK

File: hack/cherry_pick_pull.sh [hunk 1]
```diff
@@ -123,18 +123,31 @@ for pull in "${PULLS[@]}"; do
 done
 gitamcleanup=false
 
+function make-a-pr() {
+  echo "+++ Now you must propose ${NEWBRANCH} as a pull against ${BRANCH} (<--- NOT MASTER)."
+  echo "    You are constructing a pull against the upstream release branch! To do"
+  echo "    this in the GitHub UI, when you get to the 'Comparing Changes' screen, keep the"
+  echo "    'base fork' as GoogleCloudPlatform/kubernetes, but change the 'base' to e.g. 'release-1.0',"
+  echo "    presumably ${BRANCH}. (This selection is near the top left.)"
```
Comment (davidopp) [inline hack/cherry_pick_pull.sh]: I don't understand "presumably ${BRANCH}"

IIRC when I was doing this, it literally said 'release-1.0'

Comment (zmerlynn) [inline hack/cherry_pick_pull.sh]: The fixed text part is giving an example. The other part is taking a very solid guess at what you used as the thing you based this on, but in reality, if you know what you're doing, you can pass an awful lot of things to this script and it just works (and it's hard to stop you). So `${BRANCH}` could be anything from a random git commit, to a local branch, to someone's private remote branch, etc. I'm not saying you SHOULD use those, but you COULD.

Comment (davidopp) [inline hack/cherry_pick_pull.sh]: I'm still confused. What will ${BRANCH} evaluate to if you follow the steps I did earlier? Will it say 'release-1.0' or something else?

Comment (zmerlynn) [inline hack/cherry_pick_pull.sh]: Yes, it will say `release-1.0`. It looks slightly redundant.

**Comment by davidopp** [comment]
LGTM

Shippable is failing with
```
PATH=$GOPATH/bin:./third_party/etcd:$PATH ./hack/verify-gendocs.sh
docs/cluster-large.md
----
unversioned-warning:
contents were modified
FAIL: changes needed but not made due to --verify
!!! Error in ./hack/verify-gendocs.sh:51
'"${mungedocs}" "--verify=true" "--root-dir=${DOCROOT}"' exited with status 1
Call stack:
1: ./hack/verify-gendocs.sh:51 main(...)
Exiting with status 1
```

**Pull request merged**